builds/cmake/ci_build.sh

```diff
@@ -0,0 +1,264 @@
+    $CI_TIME make -j4
+    $CI_TIME make install
+    cd "${BASE_PWD}"
+fi
+if ! ((command -v dpkg-query >/dev/null 2>&1 && dpkg-query --list cxxtools-dev >/dev/null 2>&1) || \
+       (command -v brew >/dev/null 2>&1 && brew ls --versions cxxtools >/dev/null 2>&1)); then
+    $CI_TIME git clone --quiet --depth 1 -b 2.2-FTY-master https://github.com/42ity/cxxtools.git cxxtools
+    BASE_PWD=${PWD}
+    cd cxxtools
+    CCACHE_BASEDIR=${PWD}
+    export CCACHE_BASEDIR
+    git --no-pager log --oneline -n1
+    if [ -e autogen.sh ]; then
+        $CI_TIME ./autogen.sh 2> /dev/null
+    fi
+    if [ -e buildconf ]; then
+        $CI_TIME ./buildconf 2> /dev/null
+    fi
+    if [ ! -e autogen.sh ] && [ ! -e buildconf ] && [ ! -e ./configure ] && [ -s ./configure.ac ]; then
+        $CI_TIME libtoolize --copy --force && \
+        $CI_TIME aclocal -I . && \
+        $CI_TIME autoheader && \
+        $CI_TIME automake --add-missing --copy && \
+        $CI_TIME autoconf || \
+        $CI_TIME autoreconf -fiv
+    fi
+    $CI_TIME ./configure "${CONFIG_OPTS[@]}"
+    $CI_TIME make -j4
+    $CI_TIME make install
+    cd "${BASE_PWD}"
+fi
+if ! ((command -v dpkg-query >/dev/null 2>&1 && dpkg-query --list tntdb-dev >/dev/null 2>&1) || \
+       (command -v brew >/dev/null 2>&1 && brew ls --versions tntdb >/dev/null 2>&1)); then
+    $CI_TIME git clone --quiet --depth 1 -b 1.3-FTY-master https://github.com/42ity/tntdb.git tntdb
+    BASE_PWD=${PWD}
+    cd tntdb/tntdb
+    CCACHE_BASEDIR=${PWD}
+    export CCACHE_BASEDIR
+    git --no-pager log --oneline -n1
+    if [ -e autogen.sh ]; then
+        $CI_TIME ./autogen.sh 2> /dev/null
+    fi
+    if [ -e buildconf ]; then
+        $CI_TIME ./buildconf 2> /dev/null
+    fi
+    if [ ! -e autogen.sh ] && [ ! -e buildconf ] && [ ! -e ./configure ] && [ -s ./configure.ac ]; then
+        $CI_TIME libtoolize --copy --force && \
+        $CI_TIME aclocal -I . && \
+        $CI_TIME autoheader && \
+        $CI_TIME automake --add-missing --copy && \
+        $CI_TIME autoconf || \
+        $CI_TIME autoreconf -fiv
+    fi
+    $CI_TIME ./configure "${CONFIG_OPTS[@]}"
+    $CI_TIME make -j4
+    $CI_TIME make install
+    cd "${BASE_PWD}"
+fi
+
+# Build and check this project
+cd ../..
+[ -z "$CI_TIME" ] || echo "`date`: Starting build of currently tested project..."
+CCACHE_BASEDIR=${PWD}
+export CCACHE_BASEDIR
+if [ "$DO_CLANG_FORMAT_CHECK" -eq "1" ] ; then
+    PKG_CONFIG_PATH=${BUILD_PREFIX}/lib/pkgconfig $CI_TIME cmake "${CMAKE_OPTS[@]}" . \
+    && make clang-format-check || { make clang-format-diff && exit 1 ; }
+else
+    PKG_CONFIG_PATH=${BUILD_PREFIX}/lib/pkgconfig $CI_TIME cmake "${CMAKE_OPTS[@]}" .
+    $CI_TIME make all VERBOSE=1 -j4
+    $CI_TIME ctest -V
+    $CI_TIME make install
+fi
+[ -z "$CI_TIME" ] || echo "`date`: Builds completed without fatal errors!"
+
+echo "=== Are GitIgnores good after making the project '$BUILD_TYPE'? (should have no output below)"
+git status -s || true
```

Comment:
This line exposes a non-recent zproject (there is a `make` target for such sanity check now)